app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -1,13 +1,75 @@
 package org.oppia.app.parser
 
+import org.oppia.app.customview.interaction.FractionInputInteractionView
 import org.oppia.app.model.Fraction
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d+)$""".toRegex()
+  private val partialFractionOnlyRegex = """^-? ?(\d+) ?/ ?$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d+) (\d+) ?/ ?(\d+)$""".toRegex()
+  private val partialMixedNumberRegexFirst = """^-? ?(\d+) (\d+) ?$""".toRegex()
+  private val partialMixedNumberRegexSecond = """^-? ?(\d+) (\d+) ?/ ?$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  /**
+   * @param inputText is the user input in the [FractionInputInteractionView]
+   * This method helps to validate the inputText and return [FractionParsingErrors]
+   * This called on text change.
+   */
+  fun checkForErrors(inputText: String): FractionParsingErrors {
+    var rawInput: String = inputText.normalizeWhitespace()
+    if (!rawInput.matches(invalidCharsRegex))
+      return FractionParsingErrors.INVALID_CHARS
+    if (!rawInput.equals("-") && wholeNumberOnlyRegex.matchEntire(rawInput) == null &&
```

Comment:
`rawInput != "-" ` instead of `!rawInput.equals("-")`

Reply:
Updated with partial regex